Makefile.uk

```diff
@@ -133,6 +135,7 @@ LIBNEWLIBGLUE_CFLAGS-y   += $(LIBNEWLIBGLUE_SUPPRESS_FLAGS-y) $(LIBNEWLIB_FLAGS)
 LIBNEWLIBGLUE_CXXFLAGS-y += $(LIBNEWLIBGLUE_SUPPRESS_FLAGS-y) $(LIBNEWLIB_FLAGS)
 
 LIBNEWLIB_GLOBAL_SUPPRESS_FLAGS-y += -Wno-char-subscripts
+LIBNEWLIB_GLOBAL_SUPPRESS_FLAGS-y += -Wno-typedef-redefinition
```

Comment:
I get a warning on this one, using `gcc-12`
```
unrecognized command-line option ‘-Wno-typedef-redefinition’ may have been intended to silence earlier diagnostics
```
Should we check for the compiler version here?

Reply:
Alrighty, lemme run it through my compiler version gauntlet and see.

Reply:
Mkay, `-Wtypedef-redefinition` seems to be in clang since basically forever; fixing now.